### PR TITLE
[compiler] Correct dims_rhs constraint

### PR DESCRIPTION
**Pull request by gaurav1086**
Correct dims_rhs constraint

### CODE REVIEW

**Review by joker-eph — COMMENTED** [review]
Thanks for the fix!

Can you update the test coverage as well?

**Comment by gaurav1086** [comment]
@joker-eph thanks for the review. Could you kindly point me to the existing coverage/example for this (type of) functionality ?

**Comment by joker-eph** [comment]
Here is the current test https://github.com/tensorflow/tensorflow/blob/316054e5c1d71285970951dc097db4d45f1e1567/tensorflow/compiler/mlir/tensorflow/tests/einsum.mlir

**Comment by gaurav1086** [comment]
@joker-eph not familiar with the mlir format. Added a test cases (5D), please advise. thank you.

File: tensorflow/compiler/mlir/tensorflow/tests/einsum.mlir [hunk 1]
```diff
@@ -86,3 +86,10 @@ func @einsum_illegal_no_match(%arg0: tensor<4x5xf32>, %arg1: tensor<5xf32>) -> t
 // CHECK: %[[v0:.*]] = "tf.Einsum"(%arg0, %arg1) {T = "tfdtype$DT_FLOAT", equation = "ij,?zw->kq->i"} : (tensor<4x5xf32>, tensor<5xf32>) -> tensor<4xf32>
 // CHECK: return %[[v0]]
 }
+func @einsum_no_match5D(%arg0: tensor<4x5xf32>, %arg1: tensor<2x4x7x3x5xf32>) -> tensor<4xf32> {
+  %0 = "tf.Einsum"(%arg0, %arg1) {T = "tfdtype$DT_FLOAT", equation = "ij,j->i"}: (tensor<4x5xf32>, tensor<2x4x7x3x5xf32>) -> tensor<4xf32>
+  return %0 : tensor<4xf32>
+// CHECK-LABEL: einsum_no_match2
+// CHECK: %[[v0:.*]] = "tf.Einsum"(%arg0, %arg1) {T = "tfdtype$DT_FLOAT", equation = "ij,j->i"} : (tensor<4x5xf32>, tensor<5xf32>) -> tensor<4xf32>
+// CHECK: return %[[v0]]
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/tensorflow/tests/einsum.mlir:94]: You can just limit yourself to `// CHECK: %[[v0:.*]] = "tf.Einsum"` ; the rest of the line does not matter in this test I believe: we just want to make sure that Einsum isn't replaced.

File: tensorflow/compiler/mlir/tensorflow/tests/einsum.mlir [hunk 2]
```diff
@@ -86,3 +86,10 @@ func @einsum_illegal_no_match(%arg0: tensor<4x5xf32>, %arg1: tensor<5xf32>) -> t
 // CHECK: %[[v0:.*]] = "tf.Einsum"(%arg0, %arg1) {T = "tfdtype$DT_FLOAT", equation = "ij,?zw->kq->i"} : (tensor<4x5xf32>, tensor<5xf32>) -> tensor<4xf32>
 // CHECK: return %[[v0]]
 }
+func @einsum_no_match5D(%arg0: tensor<4x5xf32>, %arg1: tensor<2x4x7x3x5xf32>) -> tensor<4xf32> {
+  %0 = "tf.Einsum"(%arg0, %arg1) {T = "tfdtype$DT_FLOAT", equation = "ij,j->i"}: (tensor<4x5xf32>, tensor<2x4x7x3x5xf32>) -> tensor<4xf32>
+  return %0 : tensor<4xf32>
+// CHECK-LABEL: einsum_no_match2
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/tensorflow/tests/einsum.mlir:92]: This needs to be einsum_no_match5D

**Review by joker-eph — APPROVED** [review]

**Review by joker-eph — CHANGES_REQUESTED** [review]
(test needs update)

**Review by joker-eph — APPROVED** [review]

**Pull request merged**